autogluon/utils/tabular/ml/models/abstract/abstract_model.py

```diff
@@ -255,22 +259,41 @@ def compute_feature_importance(self, X, y, features_to_use=None, preprocess=True
 
     # TODO: Consider repeating with different random seeds and averaging to increase confidence
     # TODO: Optimize this
-    # TODO: Check memory usage to avoid OOM
     # Compute feature importance via permutation importance
     # Note: Expensive to compute
     #  Time to compute is O(predict_time*num_features)
-    def compute_permutation_importance(self, X, y, features: list, preprocess=True) -> dict:
+    def compute_permutation_importance(self, X, y, features: list, preprocess=True, silent=False) -> dict:
+        time_start = time.time()
+
+        feature_count = len(features)
+        if not silent:
+            logger.log(20, f'Computing permutation importance for {feature_count} features on {self.name} ...')
         if preprocess:
             X = self.preprocess(X)
 
-        feature_count = len(features)
+        time_start_score = time.time()
         model_score_base = self.score(X=X, y=y, preprocess=False)
+        time_score = time.time() - time_start_score
+
+        if not silent:
+            time_estimated = (feature_count + 1) * time_score + time_start_score - time_start
+            logger.log(20, f'\tExpected runtime: {time_estimated}s')
+
         model_score_diff = []
 
         row_count = X.shape[0]
         X_test_shuffled = shuffle_df_rows(X=X, seed=0)
 
-        compute_count = 200
+        X_memory_ratio_max = 0.2
+        compute_count_max = 200
+
+        X_size_bytes = sys.getsizeof(pickle.dumps(X, protocol=4))
+        available_mem = psutil.virtual_memory().available
+        X_memory_ratio = X_size_bytes / available_mem
+
+        compute_count_safe = math.floor(X_memory_ratio_max / X_memory_ratio)
+
+        compute_count = max(1, (min(compute_count_max, compute_count_safe)))
```

Comment:
This won't affect user output, right (i.e. only affects internal operations)?  Otherwise may need to add statement to documentation.

Reply:
Correct, does not alter end result, purely for speedup